File: src/main/java/com/gymtracker/gym/exerciseLogs/dto/ExerciseLogRequest.java
package com.gymtracker.gym.exerciseLogs.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ExerciseLogRequest {

    @NotNull
    private Long workoutSessionId;
    @NotBlank
    private String exerciseName;
    @Positive
    private Integer reps;
    @Positive
    private BigDecimal weightKg;
}

File: src/main/kotlin/com/gymtracker/gym/exerciseLogs/dto/ExerciseLogRequest.kt
package com.gymtracker.gym.exerciseLogs.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.PositiveOrZero
import java.math.BigDecimal

data class ExerciseLogRequest(
    val workoutSessionId: Long,
    @field:NotBlank val exerciseName: String,
    /** Row position in the client's set table. When null the server falls back to "last + 1". */
    @field:Positive val setNumber: Int?,
    @field:PositiveOrZero val reps: Int?,
    @field:PositiveOrZero val weightKg: BigDecimal?,
)
